In [1]:
import sys
sys.path.append("../")
import numpy as np
from src.hypothesis import z_test_one_sample

# Data dari Member B
n = 4042
x_bar = 0.62098  # Proporsi Pull Request yang berhasil di-merge

# Standard Deviasi Proporsi: sqrt(p * (1 - p))
sigma = np.sqrt(x_bar * (1 - x_bar))

# Nilai hipotesis awal (klaim standar 60%)
mu0 = 0.60000 

# Eksekusi fungsi Z-Test dari file hypothesis.py
hasil = z_test_one_sample(x_bar=x_bar, mu0=mu0, sigma=sigma, n=n, alternative='two-sided', alpha=0.05)

print("=== LIVE REPORT CHECKPOINT ===")
print("Z-Statistic :", round(hasil["z_stat"], 4))
print("P-Value     :", round(hasil["p_value"], 4))
print("Keputusan   :", hasil["decision"])
print("Narasi      :", hasil["interpretation"])

=== LIVE REPORT CHECKPOINT ===
Z-Statistic : 2.7494
P-Value     : 0.006
Keputusan   : Reject H0
Narasi      : With a p-value of 0.0060 < 0.05, we have sufficient evidence to reject H0 at the 0.05 significance level.


### Prosedur Uji Hipotesis (6-Step Procedure) - Member D

**Step 1: State Hypotheses**
* $H_0: \mu = 0.60$ (Proporsi pull request yang berhasil di-merge sama dengan 60%)
* $H_a: \mu \neq 0.60$ (Proporsi pull request yang berhasil di-merge secara signifikan berbeda dari 60%)

**Step 2: Assumptions Check**
* Sampel data bersifat independen yang ditarik langsung dari log historis repositori `pandas-dev/pandas`.
* Ukuran sampel sangat masif ($n = 4042 \ge 30$). Berdasarkan *Central Limit Theorem* (CLT), distribusi sampling dari rata-rata sampel dapat diasumsikan mendekati distribusi normal sehingga pendekatan uji Z-Test satu sampel ini valid digunakan.

**Step 3: Test Statistic Calculation**
* Menggunakan formula uji Z untuk proporsi sesuai dengan rujukan teori Tsun (2020):
  $$Z = \frac{\bar{x} - \mu_0}{\sigma / \sqrt{n}}$$
* Berdasarkan eksekusi modul komputasi `src/hypothesis.py`, nilai uji statistik $Z$ berhasil dikalkulasi secara presisi sebesar **2.7494**.

**Step 4: P-value and Critical Region**
* Pengujian menggunakan metode dua arah (*two-sided*) dengan tingkat signifikansi $\alpha = 0.05$. Batas wilayah kritis penolakan berdasarkan tabel distribusi normal adalah $Z < -1.96$ atau $Z > 1.96$.

**Step 5: Statistical Decision**
* Melalui komputasi berbasis paket `scipy.stats`, diperoleh *p-value* sebesar **0.0060**. Karena *p-value* ($0.0060$) < $\alpha$ ($0.05$), maka keputusan statistik yang diambil adalah **Reject $H_0$**.

**Step 6: Conclusion & Interpretation**
* Dengan tingkat signifikansi $\alpha = 0.05$, terdapat bukti statistik yang kuat untuk menyatakan bahwa proporsi pull request yang berhasil di-merge pada repositori Pandas secara signifikan berbeda dari angka asumsi 60%. Performa aktual yang menyentuh angka 62.09% merefleksikan tingkat penerimaan kontribusi kode yang sangat sehat, aktif, dan konsisten dari tim core maintainer open-source Pandas dalam mengelola proyek global.